File: models/bloodbank/CrossMatch.js
// models/CrossMatch.js
import mongoose from "mongoose";

const CrossMatchSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    requestId: { type: mongoose.Schema.Types.ObjectId, ref: "BloodRequest", required: true },
    componentId: { type: mongoose.Schema.Types.ObjectId, ref: "BloodComponent", required: true },

    result: {
      type: String,
      enum: ["Compatible", "Incompatible"],
    },

    performedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
    performedAt: Date,
  },
  { timestamps: true }
);

export default mongoose.model("CrossMatch", CrossMatchSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/CrossMatch.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "cross_matches")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CrossMatch extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "request_id", nullable = false)
    private BloodRequest request;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "component_id", nullable = false)
    private BloodComponent component;

    @Enumerated(EnumType.STRING)
    private Result result;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "performed_by")
    private User performedBy;

    @Column(name = "performed_at")
    private LocalDateTime performedAt;

    public enum Result {
        Compatible, Incompatible
    }
}
